### PR TITLE
Scheduler - introduce Scheduled#skipExecutionIf()

### PR DESCRIPTION
**Pull request by mkouba**
- the specified bean can be used to skip any execution of a
scheduled method

### CODE REVIEW

File: docs/src/main/asciidoc/scheduler-reference.adoc [hunk 1]
```diff
@@ -199,6 +199,61 @@ void nonConcurrent() {
...
+     if (service.isStarted()) {
+        // do something every second...
+     }
+   }
+}
+----
+
+The main idea is to keep the the logic to skip the execution outside the scheduled business methods so that it can be reused and refactored easily.
+
+TIP: A CDI event of type `io.quarkus.scheduler.SkippedExecution` is fired when an execution of a scheduled method is skipped.
```
Comment (machi1990) [inline docs/src/main/asciidoc/scheduler-reference.adoc:255]: Repetition from https://github.com/quarkusio/quarkus/pull/18060/files#diff-eaa32ee7e6c527977b5485533c22e2c0cd8987ae5ad4635369587568e5403363R202

Comment (machi1990) [inline docs/src/main/asciidoc/scheduler-reference.adoc:255]:
```suggestion

```

**Comment by mkouba** [comment]
In fact, one could implement the same logic with a CDI interceptor because all scheduled methods are business methods of CDI beans. However, this approach is more lightweight and brings the idea closer to the API (`io.quarkus.scheduler.Scheduled`). Also it's much easier to implement a bean than an interceptor + binding etc.

File: extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/ConditionalExecutionTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        static final AtomicBoolean DISABLED = new AtomicBoolean(true);
+
+        @Override
+        public boolean test(ScheduledExecution execution) {
+            return DISABLED.get();
+        }
+
+        void onSkip(@Observes SkippedExecution event) {
+            if (event.triggerId.equals("foo")) {
+                System.out.println(event);
```
Comment (machi1990) [inline extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/ConditionalExecutionTest.java:74]:
```suggestion
                
```

**Review by machi1990 — APPROVED** [review]
LGTM, nice addition. 

Added small suggestions.

File: docs/src/main/asciidoc/scheduler-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/scheduler-reference.adoc:255]: Yes, the idea was to make it clear that the event is fired for both "concurrent" and "skip" variants...

Comment (machi1990) [reply docs/src/main/asciidoc/scheduler-reference.adoc:255]: Right. I saw it after re-reading the section again. Let's keep it.

Comment (machi1990) [reply docs/src/main/asciidoc/scheduler-reference.adoc:255]: Can be ignored.

**Pull request merged**